### PR TITLE
ROE-1252 Api spec for OE Due Diligence

### PR DESCRIPTION
**Pull request by mattch1**
https://companieshouse.atlassian.net/browse/ROE-1252

Update api with validation for due diligence.

### CODE REVIEW

File: spec/schema.json [hunk 1]
```diff
@@ -436,29 +436,60 @@
         }
       },
       "OverseasEntityDueDiligence": {
+        "required": [
+          "name",
+          "email"
```
Comment (mouhajer-ch) [inline spec/schema.json:441]: `identity_address` is required

File: spec/schema.json [hunk 2]
```diff
@@ -436,29 +436,60 @@
...
+              {
+                "maxLength": 256
+              }
+            ]
           },
           "identity_address": {
             "$ref": "#/components/schemas/Address"
           },
           "email": {
-            "type": "string"
```
Comment (mouhajer-ch) [inline spec/schema.json:452]: Probably better if we create a component schema for email and use `$ref`

**Review by mouhajer-ch — COMMENTED** [review]
We need to update the `Address` schema to consider that the  `county` and `postcode` are required and the `property_name_number` is not in due diligence models
```
"Address": {
  "type": "object",
  ...,
  "required": [
    "line_1",
    "town",
    "country"
  ],
  "oneOf": [
    {
      "required": [
        "property_name_number"
      ]
    },
    {
      "required": [
        "county",
        "postcode"
      ]
    }
  ]
},
```

File: spec/schema.json (hunk 2 above)
Comment (mouhajer-ch) [reply spec/schema.json:452]: Lets do it when we have consistency between emails, already discussed with Francis to check why we have got some fields with length 256 and other 250 when the table can accept 256.

**Comment by mouhajer-ch** [comment]
> We need to update the `Address` schema to consider that the `county` and `postcode` are required and the `property_name_number` is not in due diligence models
> 
> ```
> "Address": {
>   "type": "object",
>   ...,
>   "required": [
>     "line_1",
>     "town",
>     "country"
>   ],
>   "oneOf": [
>     {
>       "required": [
>         "property_name_number"
>       ]
>     },
>     {
>       "required": [
>         "county",
>         "postcode"
>       ]
>     }
>   ]
> },
> ```

Fixed in the following PR https://github.com/companieshouse/overseas-entities-api/pull/93

**Review by markpit — APPROVED** [review]

**Pull request merged**